contracts/ronin/slash-indicator/SlashFastFinality.sol

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.9;
+
+import "../../interfaces/slash-indicator/ISlashFastFinality.sol";
+import { IRoninValidatorSet } from "../../interfaces/validator/IRoninValidatorSet.sol";
+import { IProfile } from "../../interfaces/IProfile.sol";
+import { IRoninTrustedOrganization } from "../../interfaces/IRoninTrustedOrganization.sol";
+import "../../precompile-usages/PCUValidateFastFinality.sol";
+import "../../extensions/collections/HasContracts.sol";
+import "../../utils/CommonErrors.sol";
+
+abstract contract SlashFastFinality is ISlashFastFinality, HasContracts, PCUValidateFastFinality {
+  /// @dev The amount of RON to slash fast finality.
+  uint256 internal _slashFastFinalityAmount;
+  /// @dev The block number that the punished validator will be jailed until, due to malicious fast finality.
+  uint256 internal _fastFinalityJailUntilBlock;
+  /// @dev Recording of submitted proof to prevent relay attack.
+  mapping(bytes32 => bool) internal _processedEvidence;
+
+  /**
+   * @dev This empty reserved space is put in place to allow future versions to add new
+   * variables without shifting down storage in the inheritance chain.
+   */
+  uint256[21] private ______gap;
+
+  modifier onlyGoverningValidator() {
+    if (_getGovernorWeight(msg.sender) == 0) revert ErrUnauthorized(msg.sig, RoleAccess.GOVERNOR);
+    _;
+  }
+
+  /**
+   * @inheritdoc ISlashFastFinality
+   */
+  function slashFastFinality(
+    address consensusAddr,
+    bytes memory voterPublicKey,
+    uint256 targetBlockNumber,
+    bytes32[2] memory targetBlockHash,
+    bytes[][2] memory listOfPublicKey,
+    bytes[2] memory aggregatedSignature
+  ) external override onlyGoverningValidator {
+    IProfile profileContract = IProfile(getContract(ContractType.PROFILE));
+    bytes memory expectingPubKey = (profileContract.getId2Profile(consensusAddr)).pubkey;
+    if (keccak256(voterPublicKey) != keccak256(expectingPubKey)) revert ErrInvalidArguments(msg.sig);
+
+    bytes32 evidenceHash = keccak256(abi.encodePacked(consensusAddr, targetBlockNumber));
```

Comment:
should use `abi.encode` instead of `abi.encodePacked`
Ref: [solcurity#C12](https://github.com/transmissions11/solcurity)

Reply:
No need, two params are static type, safe for using `abi.encodePacked`.